snowshu/core/docker.py

```diff
@@ -31,24 +31,28 @@ def _create_snowshu_volume(self, volume_name: str) -> docker.models.volumes.Volu
     def convert_container_to_replica(
             self,
             replica_name: str,
-            container: docker.models.containers.Container) -> docker.models.images.Image:
+            active_container: docker.models.containers.Container,
+            passive_container: docker.models.containers.Container) -> None:
         """coerces a live container into a replica image and returns the image.
 
         replica_name: the name of the new replica
         """
-        replica_name = self.sanitize_replica_name(replica_name)
-        logger.info(f'Creating new replica image with name {replica_name}...')
-        try:
-            self.client.images.remove(replica_name, force=True)
-        except docker.errors.ImageNotFound:
-            pass
-        replica = container.commit(
-            repository=self.sanitize_replica_name(replica_name)
-        )
-        logger.info(f'Replica image {replica.tags[0]} created. Cleaning up...')
-        self.remove_container(container.name)
-
-        return replica
+        container_list = [
+            active_container, passive_container] if passive_container else [active_container]
+        for container in container_list:
+            container.start()
```

Comment:
Why do we need to restart each container here?

Reply:
Because active container is committed and removed by this point, and passive one is stopped so it cannot be committed if it is not running. Trying to start a running container does nothing and returns None.